docs/design/sync-manager.md

```diff
@@ -269,13 +273,181 @@ manager, we have the following hard requirements:
+"choose what to sync" on the device. This might be surprising, and possibly
+contentious, but should be very rare.
+
+#### Examples
+
+(Note that for simplicity, this section uses JSON to represent the declined
+list, but that's not how it will be specified in practice)
+
+Let's take Fenix as an example, when a user signs in to an existing account
+and before they've explicitly chosen what to sync on that device:
+
+* On the very first sync, it will supply `{"bookmarks": Unknown, "history", Unknown}`
+* When sync returns, it might get back, for example.
+  `{"bookmarks": Enabled, "history": Disabled, "passwords": Enabled, ...}`
+* The application is expected to persist *all* of these values - including 
+  "passwords", even though it doesn't understand that collection. This value
+  may be used to repopulate the declined list should this device be the first to
+  sync after storage is reset.
+
+For any app on which an account is created, and where a "choose what to sync"
+UI is part of the account creation flow:
+
+* On the very first sync, it will supply all of the "offered" engines, and
+  will supply explicit "Enabled" or "Disabled" values - nothing will be specified
+  as "Unknown".
+* Because this will be the first sync for the account, no other engines will
+  exist on the server, so the exact same states will be returned after the sync.
+
+And some edge-cases:
+
+* On the very first sync, an app supplies
+  `{"bookmarks": Unknown, "history", Unknown, "not-an-engine": Unknown}`
+* If the sync fails early, it gets the exact same thing back.
+* If the sync works, it gets back, say:
+  `{"not-an-engine": Unknown, "bookmarks": Enabled, "history": Disabled, "passwords": Enabled, ...}`
+
+#### How the sync manager resolves these
+
+There's no canonical list of engines, so somehow the sync manager needs to work
+out what to do. When storage already exists on the server, it's relatively simple -
+we have `declined` in meta/global and we have the list of current collections in
+info/collections.
+
+If there is no storage things get tricky. Consider the Fenix example above -
+it will find no storage, and even though it doesn't sync passwords, it knows
+passwords should be enabled. It will not put passwords in the declined list, but
+neither will it create the passwords collection. So even though the app supplied
+`{passwords: Enabled}`, it will now get back `{passwords: Unknown}`. This will
+only be a problem is Fenix decides to show "passwords" in its "choose what to
+sync" - so Fenix should probably avoid offering unknown engines (which is fine,
+because that would probably confuse the user anyway!) Regardless, in this
+scenario, some other device that *does* sync passwords will show up and will
+remember that it was enabled, so everything ends up OK.
+
+#### How do new engines become enabled?
+
+As implied above, the state of an engine transitions from "Unknown" to "Allowed"
```

Comment:
So, I guess the hypothetical "accepted" list you mentioned earlier would, if present, be a list of engines whose values would never be unknown?

Reply:
They would never be *returned* as unknown. For a first sync on a device where CWTS is not offered, they may be *supplied* as unknown though. Does that clarify?